<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/asil/rossi_workspace/metu-ros-pkg/stacks/aff_learning/humanoid_aff_learning/icub_aff_learning/al_robots/defs/icub.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="icub" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" xmlns:xacro="http://ros.org/wiki/xacro">
  <link name="torso">
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0.0 -0.08"/>
      <geometry>
        <box size="0.2 0.16 0.26"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 0.0 -0.08"/>
      <geometry>
        <box size="0.2 0.16 0.26"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="torso">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <!-- <xacro:shape_sphere name="head" r="0.10" xyz="0 0 0" rpy="0 0 0" mass="5" material="Gazebo/FlatWhite"> -->
  <!--   <xacro:blue /> -->
  <!-- </xacro:shape_sphere> -->
  <!-- <joint name="base_to_right_leg" type="fixed"> -->
  <!--   <parent link="torso"/> -->
  <!--   <child link="head"/> -->
  <!--   <origin xyz="0.22 0 .25"/> -->
  <!-- </joint> -->
  <!-- <xacro:shape_cylinder name="neck" l="0.28" r="0.05" xyz="0 0 0" rpy="${M_PI/2} 0 0" mass="5" material="Gazebo/FlatBlack"> -->
  <!--   <xacro:black /> -->
  <!-- </xacro:shape_cylinder> -->
  <link name="left_shoulder">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
      <xacro:blue/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="5"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="left_shoulder">
    <material>Gazebo/FlatWhite</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="right_shoulder">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
      <xacro:blue/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="5"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="right_shoulder">
    <material>Gazebo/FlatWhite</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="right_upper_arm">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.1"/>
      <geometry>
        <cylinder length="0.15" radius="0.02"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.1"/>
      <geometry>
        <cylinder length="0.15" radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="5"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="right_upper_arm">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="left_upper_arm">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.1"/>
      <geometry>
        <cylinder length="0.15" radius="0.02"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.1"/>
      <geometry>
        <cylinder length="0.15" radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="5"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="left_upper_arm">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="left_elbow">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
      <xacro:blue/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="5"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="left_elbow">
    <material>Gazebo/FlatWhite</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="right_elbow">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
      <xacro:blue/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="5"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="right_elbow">
    <material>Gazebo/FlatWhite</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="left_lower_arm">
    <visual>
      <origin rpy="1.570795 0 0" xyz="0  -0.1 0"/>
      <geometry>
        <cylinder length="0.2" radius="0.02"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
    </visual>
    <collision>
      <origin rpy="1.570795 0 0" xyz="0  -0.1 0"/>
      <geometry>
        <cylinder length="0.2" radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="5"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="left_lower_arm">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="right_lower_arm">
    <visual>
      <origin rpy="1.570795 0 0" xyz="0 0.1 0"/>
      <geometry>
        <cylinder length="0.2" radius="0.02"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
    </visual>
    <collision>
      <origin rpy="1.570795 0 0" xyz="0 0.1 0"/>
      <geometry>
        <cylinder length="0.2" radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="5"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="right_lower_arm">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="left_hand">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
      <xacro:blue/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="left_hand">
    <material>Gazebo/FlatWhite</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="right_hand">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
      <xacro:blue/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.03"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="right_hand">
    <material>Gazebo/FlatWhite</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <!-- I guess the order is pitch, roll, yaw. And it really matters -->
  <!-- It should be roll pitch yaw as rpy suggests-->
  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="base_link">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="torso_pitch">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="torso_pitch">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="torso_roll">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="torso_roll">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="torso_yaw">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="torso_yaw">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="right_shoulder_pitch">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="right_shoulder_pitch">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="right_shoulder_roll">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="right_shoulder_roll">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="right_shoulder_yaw">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="right_shoulder_yaw">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="right_wrist_pitch">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="right_wrist_pitch">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="right_wrist_yaw">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="right_wrist_yaw">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="right_wrist_yaw_2">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="right_wrist_yaw_2">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="left_shoulder_pitch">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="left_shoulder_pitch">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="left_shoulder_roll">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="left_shoulder_roll">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="left_shoulder_yaw">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="left_shoulder_yaw">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="left_wrist_pitch">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="left_wrist_pitch">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="left_wrist_yaw">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="left_wrist_yaw">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <link name="left_wrist_yaw_2">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <material name="black">
        <color rgba="0.1 0.1 0.1 0.9"/>
      </material>
      <!--material name="red">
	    <color rgba="0.8 0 0 1"/>
	    </material-->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="left_wrist_yaw_2">
    <material>Gazebo/FlatBlack</material>
    <selfCollide>false</selfCollide>
    <turnGravityOff>True</turnGravityOff>
  </gazebo>
  <joint name="base_dummy_0" type="fixed">
    <parent link="base_link"/>
    <child link="torso_pitch"/>
    <origin rpy="1.570795 0 0" xyz="0 0 0"/>
    <!-- <origin xyz="0 0 0" rpy="${-M_PI/2} ${-M_PI/2} ${M_PI}" /> -->
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="torso_0" type="continuous">
    <parent link="torso_pitch"/>
    <child link="torso_roll"/>
    <origin rpy="0 0 1.570795" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="torso_1" type="continuous">
    <parent link="torso_roll"/>
    <child link="torso_yaw"/>
    <origin rpy="1.570795 0 0" xyz="0.032 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="torso_2" type="continuous">
    <parent link="torso_yaw"/>
    <child link="torso"/>
    <origin rpy="1.570795 0 -1.570795" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <!-- Right Arm -->
  <joint name="right_arm_dummy_0" type="fixed">
    <parent link="torso"/>
    <child link="right_shoulder_pitch"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="right_arm_0" type="continuous">
    <parent link="right_shoulder_pitch"/>
    <child link="right_shoulder_roll"/>
    <origin rpy="1.570795 0 -1.83259416667" xyz="-0.02336 0 -0.1433"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="right_arm_1" type="continuous">
    <parent link="right_shoulder_roll"/>
    <child link="right_shoulder_yaw"/>
    <origin rpy="1.570795 0 -1.570795" xyz="0 0 -0.10774"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="right_arm_2" type="continuous">
    <parent link="right_shoulder_yaw"/>
    <child link="right_shoulder"/>
    <origin rpy="-1.570795 0 -1.570795" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="right_arm_dummy_1" type="fixed">
    <parent link="right_shoulder"/>
    <child link="right_upper_arm"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="right_arm_3" type="continuous">
    <parent link="right_upper_arm"/>
    <child link="right_lower_arm"/>
    <origin rpy="-1.570795 0 -1.83259416667" xyz="0 0 -0.15228"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="right_arm_dummy_2" type="fixed">
    <parent link="right_lower_arm"/>
    <child link="right_elbow"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="right_arm_4" type="continuous">
    <parent link="right_elbow"/>
    <child link="right_wrist_pitch"/>
    <origin rpy="1.570795 0 0" xyz="0.015 0  0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="right_arm_5" type="continuous">
    <parent link="right_wrist_pitch"/>
    <child link="right_wrist_yaw"/>
    <origin rpy="1.570795 0 -1.570795" xyz="0 0 -0.1373"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="right_arm_6" type="continuous">
    <parent link="right_wrist_yaw"/>
    <child link="right_wrist_yaw_2"/>
    <origin rpy="0 0 -1.570795" xyz="0 0 0 "/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="right_arm_7" type="continuous">
    <parent link="right_wrist_yaw_2"/>
    <child link="right_hand"/>
    <origin rpy="1.570795 0 0" xyz="0.0625 0 0.016"/>
    <axis xyz="0 0 1"/>
  </joint>
  <!-- left Arm -->
  <joint name="left_arm_dummy_0" type="fixed">
    <parent link="torso"/>
    <child link="left_shoulder_pitch"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="left_arm_0" type="continuous">
    <parent link="left_shoulder_pitch"/>
    <child link="left_shoulder_roll"/>
    <origin rpy="-1.570795 0 1.83259416667" xyz="0.02336 0 -0.1433"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="left_arm_1" type="continuous">
    <parent link="left_shoulder_roll"/>
    <child link="left_shoulder_yaw"/>
    <origin rpy="-1.570795 0 1.570795" xyz="0 0 0.10774"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="left_arm_2" type="continuous">
    <parent link="left_shoulder_yaw"/>
    <child link="left_shoulder"/>
    <origin rpy="1.570795 0 -1.570795" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="left_arm_dummy_1" type="fixed">
    <parent link="left_shoulder"/>
    <child link="left_upper_arm"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="left_arm_3" type="continuous">
    <parent link="left_upper_arm"/>
    <child link="left_lower_arm"/>
    <origin rpy="-1.570795 0 1.30899583333" xyz="0 0 0.15228"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="left_arm_dummy_2" type="fixed">
    <parent link="left_lower_arm"/>
    <child link="left_elbow"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="left_arm_4" type="continuous">
    <parent link="left_elbow"/>
    <child link="left_wrist_pitch"/>
    <origin rpy="1.570795 0 0" xyz="-0.015 0  0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="left_arm_5" type="continuous">
    <parent link="left_wrist_pitch"/>
    <child link="left_wrist_yaw"/>
    <origin rpy="1.570795 0 -1.570795" xyz="0 0 0.1373"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="left_arm_6" type="continuous">
    <parent link="left_wrist_yaw"/>
    <child link="left_wrist_yaw_2"/>
    <origin rpy="0 0 -1.570795" xyz="0 0 0 "/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="left_arm_7" type="continuous">
    <parent link="left_wrist_yaw_2"/>
    <child link="left_hand"/>
    <origin rpy="1.570795 0 0" xyz="-0.0625 0 0.016"/>
    <axis xyz="0 0 1"/>
  </joint>
</robot>
